### DISCUSSION TITLE
Using serial?

### DISCUSSION THREAD

**Discussion by coderatchance** [SpenceKonde/megaTinyCore#1110] [Library discussion]
Hello all,

so far I have been using dedicated serial via the Pb3 and Pb4 on the ATTiny3216, all good, however, I am looking to move the serial, namely rx, tx to the alternative pins, such as Pa2 and Pa1. This is one option, however, there is as well software serial. for which I have already found instructions... so what would you suggest me to use? Would either work, is either useful at all? I guess the general SoftwareSerial from Arduino works?

Thank you for your comments, ideas.

Best.

**Comment by hmeijdam** [comment]
Softwareserial takes more processing time, so if you have a demanding or timing sensitive application, I would stay away from softwareserial. It can also not receive and send at the same time.

**Comment by coderatchance** [comment]
Hand thank you! I was thinking this may be the case. Where can we setup the ports in the megaTinyCore when the alternative pins are used? Best. On 10 Jun 2024, at 14:37, Hans ***@***.***> wrote:﻿
Softwareserial takes more processing time, so if you have a demanding or timing sensitive application, I would stay away from softwareserial. It can also not receive and send at the same time.

**Comment by hmeijdam** [comment]
You can try this.

```
void setup() {
}

void loop() {
Serial.begin(115200);
Serial.println("This prints on the default UART pins");
Serial.end();
delay(1000);
Serial.swap(1);// change to alternate UART pins
Serial.begin(9600);
Serial.println("This prints on the alternate UART pins");
Serial.end();
delay(1000);
Serial.swap(0);// change to default UART pins
}
```

**Reply by coderatchance** [reply]
Works, thanks!
